src/os/bluestore/BlueFS.cc

```diff
@@ -1224,7 +1226,10 @@ void BlueFS::_compact_log_async(std::unique_lock<std::mutex>& l)
 
   // delete the new log, remove from the dirty files list
   _close_writer(new_log_writer);
-  dirty_files.erase(dirty_files.iterator_to(*new_log));
+  if (dirty_files.count(new_log->dirty_seq)) {
```

Comment:
I think this should be if (new_log->dirty_seq).  if that is non-zero, then we know the matching dirty_files list exists for that seq, right?